Mathlib/LinearAlgebra/Matrix/Charpoly/Basic.lean

```diff
@@ -4,6 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Kim Morrison
 -/
 import Mathlib.LinearAlgebra.Matrix.Adjugate
+import Mathlib.LinearAlgebra.Matrix.Block
```

Comment:
This seems to be somewhat heavy; I won't insist on it, but please consider whether it's worth putting these results in a new file.

Reply:
Hmm, I'm not convinced that there's enough to say about block triangular and characteristic matrices to deserve a new file, and it'd be much heavier to put these things in the block matrix file. @eric-wieser what do you think?

Reply:
Oof the import is indeed heavier than I thought. I still don't expect there's enough material about these to deserve a new file, but I'll add a comment suggesting that it should be split out if something else is thought of in the future. How does that sound to you?